src/satellite/src/satellite.js

```diff
@@ -78,10 +76,12 @@ class Satellite {
 
     if (!this.server) {
       finished();
-      return;
+      return null;
```

Comment:
You're mixing two calling patterns here: callbacks and Promises.  You have to pick one, and I'd suggest using Promises.  This means that every `return` has to be a `Promise`.

You need to modify `stop` like so:

```js
const { promisify } = require('util');
...
stop() {
  const server = this.server;

  // Convert server.close() to return a Promise
  const promisifiedClose = server ? promisify(server.close) : Promise.resolve;

  // Shut down all long-running connections
  return Promise.all([this.shutDown(), redis.shutDown(), elastic.shutDown(), promisifiedClose()]);
}
```

Untested code, but that's the idea.

Reply:
Oh, promisify would help me